src/multi_vector_simulator/C1_verification.py

```diff
@@ -766,3 +770,62 @@ def check_for_sufficient_assets_on_busses(dict_values):
                 f"so that the bus is not a dead end should be two, excluding the excess sink. "
                 f"These are the connected assets: {asset_string}"
             )
+
+
+def check_energy_system_can_fulfill_max_demand(dict_values):
+    r"""
+    Logs a logging.warning message if the aggregated installed capacity and maximum capacity (if applicable)
+    of all conversion, generation and storage assets connected to one bus is smaller than the maximum demand.
+    
+   The check is applied to each bus of the energy system. Check passes when the potential peak supply is
+   larger then or equal to the peak demand on the bus, or if the maximum capacity of an asset is set to
+   None for optimization.
+
+    Parameters
+    ----------
+    dict_values : dict
+        Contains all input data of the simulation.
+
+    Returns
+    -------
+    Indirectly, logs a logging.warning message if the installed and maximum capacities of
+    conversion/generation/storage assets are less than the maximum demand, for each bus.
+
+    Notes
+    -----
+    Tested with:
+    - test_check_energy_system_can_fulfill_max_demand_sufficient_capacities()
+    - test_check_energy_system_can_fulfill_max_demand_no_maximum_capacity()
+    - test_check_energy_system_can_fulfill_max_demand_insufficient_capacities()
+    - test_check_energy_system_can_fulfill_max_demand_with_storage()
+    """
+    skip_check = False
+    for bus in dict_values[ENERGY_BUSSES]:
+        for item in dict_values[ENERGY_BUSSES][bus][ASSET_DICT]:
+            peak_demand = 0
+            peak_generation = 0
+            # filters out excess energy sinks, leaving only actual demand profiles
+            if item in dict_values[ENERGY_CONSUMPTION] and \
+                    dict_values[ENERGY_CONSUMPTION][item][DISPATCHABILITY] is False:
+                peak_demand += dict_values[ENERGY_CONSUMPTION][item][TIMESERIES_PEAK]
+            for group in [ENERGY_CONVERSION, ENERGY_PRODUCTION]:
+                if item in dict_values[group]:
+                    peak_generation += dict_values[group][item][INSTALLED_CAP][VALUE]
+                    if dict_values[group][item][MAXIMUM_CAP][VALUE] is None:
+                         skip_check = True
+                    else:
+                         peak_generation += dict_values[group][item][MAXIMUM_CAP][VALUE]
+            if item in dict_values[ENERGY_STORAGE]:
+                peak_generation += dict_values[ENERGY_STORAGE][item][OUTPUT_POWER][INSTALLED_CAP][VALUE]
+            elif dict_values[ENERGY_STORAGE][item][OPTIMIZE_CAP][VALUE] is True:
+                # unlike the conversion/production assets, no maximum capacities are defined for
+                # storage assets and therefore as soon as a storage asset is connected, the check
+                # should pass
+                skip_check = True
+
+    if peak_generation < peak_demand and skip_check is False:
```

Comment:
Do you plan to apply the test to each individual bus (easier to debug) or for all busses at once? 

If the first, which I would prefer, I think you need to indent this to apply the test to each individual bus (keep it in the bus-loop started in l. 803) - right now you apply it to all busses at once and if there is a `maximumCap` anywhere, the tests pass. If they fail, the error message if logged for the last bus only (even if that may not be the problem child).

`skip_check = False` in line 802 would also have to be moved into the loop of the busses (line 803).